testing/testcafe/tests/editors/overlays/scrolling.ts

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable no-multi-str */
+import { ClientFunction, Selector } from 'testcafe';
+import { createScreenshotsComparer } from 'devextreme-screenshot-comparer';
+import { testScreenshot, isMaterial } from '../../../helpers/themeUtils';
+import url from '../../../helpers/getPageUrl';
+import createWidget from '../../../helpers/createWidget';
+import {
+  appendElementTo, getComputedPropertyValue, getDocumentScrollTop, insertStylesheetRulesToPage,
+  removeAttribute,
+} from '../../../helpers/domUtils';
+import Popup from '../../../model/overlay/index';
+
+fixture`Popup scrolling`
+  .page(url(__dirname, '../../container.html'));
+
+if (!isMaterial()) {
+  [false, true].forEach((shading) => {
+    [false, true].forEach((enableBodyScroll) => {
+      [false, true].forEach((fullScreen) => {
+        test(`Overlay native scrolling, shading: ${shading}, enableBodyScroll: ${enableBodyScroll}, fullScreen: ${fullScreen}`, async (t) => {
+          const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+          const overlay = new Popup('#overlay');
+
+          const body = Selector('body');
+
+          const checkBodyStyles = async ({ paddingRight, overflow }) => {
+            await t
+              .expect(getComputedPropertyValue('body', 'padding-right'))
+              .eql(paddingRight)
+              .expect(getComputedPropertyValue('body', 'overflow'))
+              .eql(overflow)
+              .expect(getComputedPropertyValue('body', 'position'))
+              .eql('static')
+              .expect(getComputedPropertyValue('body', 'top'))
+              .eql('auto')
+              .expect(getComputedPropertyValue('body', 'left'))
+              .eql('auto');
+          };
+
+          await t
+            .scroll(body, 300, 0);
+
+          await checkBodyStyles({ paddingRight: '0px', overflow: 'visible' });
+          await t
+            .expect(getDocumentScrollTop())
+            .eql(258);
```

Comment:
This magic number is strange to me. We scroll on 300, but get a scrollTop 258. I'm afraid it can make this test unstable in the future.
Maybe it's better to use `dispatchEvent` or some other way to get rid of this?

Reply:
Yes, we may. I used another way for it